tests/cupyx_tests/scipy_tests/interpolate_tests/test_ppoly.py

```diff
@@ -0,0 +1,766 @@
+        ip = p.antiderivative()
+
+        # check continuity
+        testing.assert_allclose(ip(0.5 - 1e-9), ip(0.5 + 1e-9), rtol=1e-8)
+
+        # check that only lowest order coefficients were changed
+        p2 = ip.derivative()
+        testing.assert_allclose(p2.c, p.c)
+
+    def test_integrate(self):
+        numpy.random.seed(1234)
+        x = numpy.sort(numpy.r_[0, numpy.random.rand(11), 1])
+        y = numpy.random.rand(len(x))
+
+        spl = sc_interpolate.splrep(x, y, s=0, k=5)
+        cp_spl = tuple([cupy.asarray(x) if isinstance(x, np.ndarray) else x
+                        for x in spl])
+        # spl_cupy = (cupy.asarray(spl[0]), cupy.asarray(spl[1]), spl[2])
+        pp = cupyx.scipy.interpolate.PPoly.from_spline(cp_spl)
+
+        a, b = 0.3, 0.9
+        ig = pp.integrate(a, b)
+
+        # ipp = pp.antiderivative()
+        # testing.assert_allclose(ig, ipp(b) - ipp(a))
+        testing.assert_allclose(ig, sc_interpolate.splint(a, b, spl))
+
+        a, b = -0.3, 0.9
+        ig = pp.integrate(a, b, extrapolate=True)
+        # testing.assert_allclose(ig, ipp(b) - ipp(a))
+
+        assert (cupy.isnan(pp.integrate(a, b, extrapolate=False)).all())
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_integrate_readonly(self, xp, scp):
+        x = xp.array([1, 2, 4])
+        c = xp.array([[0., 0.], [-1., -1.], [2., -0.], [1., 2.]])
+
+        ret = []
+        for writeable in (True, False):
+            x.flags.writeable = writeable
+
+            P = scp.interpolate.PPoly(c, x)
+            vals = P.integrate(1, 4)
+            ret.append(vals)
+
+        return ret
+
+    def test_integrate_periodic(self):
+        x = cupy.array([1, 2, 4])
+        c = cupy.array([[0., 0.], [-1., -1.], [2., -0.], [1., 2.]])
+
+        P = cupyx.scipy.interpolate.PPoly(c, x, extrapolate='periodic')
+        poly_int = P.antiderivative()
+
+        period_int = poly_int(4) - poly_int(1)
+
+        assert_allclose(P.integrate(1, 4), period_int)
+        assert_allclose(P.integrate(-10, -7), period_int)
+        assert_allclose(P.integrate(-10, -4), 2 * period_int)
+
+        assert_allclose(P.integrate(1.5, 2.5), poly_int(2.5) - poly_int(1.5))
+        assert_allclose(
+            P.integrate(3.5, 5), poly_int(2) - poly_int(1) +
+            poly_int(4) - poly_int(3.5))
+        assert_allclose(P.integrate(3.5 + 12, 5 + 12),
+                        poly_int(2) - poly_int(1) + poly_int(4) -
+                        poly_int(3.5))
+        assert_allclose(P.integrate(3.5, 5 + 12),
+                        poly_int(2) - poly_int(1) + poly_int(4) -
+                        poly_int(3.5) + 4 * period_int)
+
+        assert_allclose(P.integrate(0, -1), poly_int(2) - poly_int(3))
+        assert_allclose(P.integrate(-9, -10), poly_int(2) - poly_int(3))
+        assert_allclose(
+            P.integrate(0, -10), poly_int(2) - poly_int(3) - 3 * period_int)
+
+    # TODO: Add a solve implementation
```

Comment:
Maybe better `pytest.skip` this and other tests of roots and solve methods? While at it, maybe the the comment/skip message could be made more specific and point to the lack of non-symmetric eigenvalue solver in CUDA.